api/src/main/java/org/openmrs/module/queue/api/impl/QueueEntryServiceImpl.java

```diff
@@ -225,4 +250,18 @@ private void endQueueEntry(@NotNull QueueEntry queueEntry) {
 		queueEntry.setEndedAt(new Date());
 		dao.createOrUpdate(queueEntry);
 	}
+	
+	@Override
+	public QueueEntry getPreviousQueueEntry(@NotNull QueueEntry queueEntry) {
+		QueueEntrySearchCriteria criteria = new QueueEntrySearchCriteria();
+		criteria.setPatient(queueEntry.getPatient());
```

Comment:
The `Queue` also needs to match.  That appears to be missing here.

Reply:
Wait... that's not true. You can transition to a different queue.

Reply:
Oh right, good point.

Reply:
Unresolving this.  You are right, you can transition to a different queue, but we do track the `queueComingFrom` on the new Queue Entry.  So in this case, we should set `criteria.setQueue(queueEntry.getQueueComingFrom())` if `queueEntry.getQueueComingFrom() != null`